server/cluster/cluster.go

```diff
@@ -1094,27 +1106,24 @@ func (c *RaftCluster) AttachAvailableFunc(storeID uint64, limitType storelimit.T
 	c.core.AttachAvailableFunc(storeID, limitType, f)
 }
 
-// SetStoreState sets up a store's state.
-func (c *RaftCluster) SetStoreState(storeID uint64, state metapb.StoreState) error {
+// UpStore up a store from offline
+func (c *RaftCluster) UpStore(storeID uint64) error {
 	c.Lock()
 	defer c.Unlock()
-
 	store := c.GetStore(storeID)
 	if store == nil {
 		return errs.ErrStoreNotFound.FastGenByArgs(storeID)
 	}
-
-	if store.GetState() == metapb.StoreState_Tombstone && state != metapb.StoreState_Tombstone {
-		if err := c.checkStoreVersion(store.GetMeta()); err != nil {
-			return err
-		}
+	if store.IsTombstone() || store.IsPhysicallyDestoryAndOffline() {
+		return errs.ErrStoreTombstone.FastGenByArgs(storeID)
 	}
 
-	newStore := store.Clone(core.SetStoreState(state))
-	log.Warn("store update state",
+	newStore := store.Clone(core.SetStoreState(metapb.StoreState_Up))
+	log.Warn("up store",
 		zap.Uint64("store-id", storeID),
-		zap.Stringer("new-state", state))
+		zap.Stringer("old-state", store.GetState()))
```

Comment:
How about printing both the old and the new state?

Reply:
The information of the new state is already in the log's prefix  `up store..`?

Reply:
I prefer to record them not only in the log message